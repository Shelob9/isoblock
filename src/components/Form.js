import React from "react";

/**
 * The Form component
 */
export const Form = ({ defaultValue, onSubmit, className, id, onChange }) => {
  return (
    <form onSubmit={onSubmit} className={className}>
      <div>
        <label htmlFor={id}>Form Field</label>
        <input
          value={defaultValue}
          id={id}
          onChange={onChange}
          className={"isoblock-field"}
        />
      </div>
    </form>
  );
};
