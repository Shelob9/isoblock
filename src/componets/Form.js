
export const Form = ({defaultValue,onSubmit,className, id,onChange} ) => {
    return (
        <form onSubmit={onSubmit} className={className}>
            <div>
                <label htmlFor={id}>
                    Form Field
                </label>
                <input
                    value={defaultValue}
                    id={id}
                    onChange={onChange}
                    className={'hydrate-test-field'}
                />
            </div>
        </form>
    );
};
