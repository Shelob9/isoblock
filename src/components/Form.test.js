//
import React from "react";

//Import component to test
import { Form } from "./Form";

//Import a renderer to test with
import TestRenderer from "react-test-renderer";

test("Form component", () => {
  //Props with sample data
  const props = {
    defaultValue: "Tooths",
    className: "wp-blocks-tooths",
    id: "something",
  };

  const component = TestRenderer.create(<Form {...props} />);
  expect(component.toJSON()).toMatchSnapshot();
});
