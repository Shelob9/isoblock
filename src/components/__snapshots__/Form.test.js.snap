// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component 1`] = `
<form
  className="wp-blocks-tooths"
>
  <div>
    <label
      htmlFor="something"
    >
      Form Field
    </label>
    <input
      className="isoblock-field"
      id="something"
      value="Tooths"
    />
  </div>
</form>
`;
